pkg/sensors/sensors_windows.go

```diff
@@ -0,0 +1,314 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package sensors
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/cilium/tetragon/pkg/bpf"
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/pkg/policyfilter"
+	"github.com/cilium/tetragon/pkg/sensors/program"
+	"github.com/cilium/tetragon/pkg/tracingpolicy"
+
+	// load rthooks for policy filter
+	_ "github.com/cilium/tetragon/pkg/policyfilter/rthooks"
+)
+
+var (
+	// allPrograms are all the loaded programs. For use with Unload().
+	allPrograms = []*program.Program{}
+	// allPrograms lock
+	allProgramsMutex sync.Mutex
+	// AllMaps are all the loaded programs. For use with Unload().
+	AllMaps = []*program.Map{}
+)
+
+// Sensors
+//
+// Sensors are a mechanism for dynamically loading/unloading bpf programs.
+// Contrarily to low-level facilities like kprobes, sensors are meant to be
+// visible to end users who can enable/disable them.
+//
+// Sensor control operations are done in a separate goroutine which acts as a
+// serialization point for concurrent client requests.
+
+// Sensor is a set of BPF programs and maps that are managed as a unit.
+type Sensor struct {
```

Comment:
A lot of the code here seems to be the same as the Linux version. The `Sensor` struct seems to be an example. Is there a reason for this code to be duplicated? It's a good idea to avoid code duplication as much as possible.

Reply:
yes, when you make a diff of windows and linux version it's just `standardTypes` map plus one other change that suggests that the sensor.go used for windows version is missing last upstream change:
 4f6f97cb6b6d tetragon: Pass sensor maps to program loader

Reply:
About code duplication. yes, there is a lot of common code that is duplicated. I have duplicated code and created separate _linux.go files where I think we will need to make big changes based on how ebpf on windows evolves. This include structures, enums and  `BPF_PROG_TYPE` constants. Windows has it's own `BPF_PROG_TYPE` consts, which may or may not have the same values. 
I have prioritized not making any changes to _linux code further down the road, as compared to making frequent changes in common code. We can move common code in common files when we find the 'real common code' between the two OS. 

But I am very happy to merge common code if team prioritizes code reuse.

Reply:
I have merged latest upstream changes in Windows.